### DISCUSSION TITLE
object has no attribute

### DISCUSSION THREAD

**Discussion by mamax1101** [gijzelaerr/python-snap7#395] [Q&A] [answered]
Hello, 
I am making a visualization program with the library in parallel I my small test codes. I have to switch to mod
Start and Stop my cpu but each time I have an error message. 
`'Client' object has no attribute 'CpuStop'`

I don't really understand where it comes from, my plc is well attributed. a little idea of ​​the problem. I looked in the so I found nothing. thank you

```
import snap7

plc = snap7.client.Client()
plc.connect("210.125.55.106", 0, 1)

plc.CpuStop()
```

**Comment by swamper123** [comment, marked as answer]
Hey ho,

well there is no Client method "CpuStop". Just this one:

https://github.com/gijzelaerr/python-snap7/blob/076dbee171639b27d7179c9ce69f9444f5669067/snap7/client.py#L93

**Reply by mamax1101** [reply]
strange, on the snap7 site we know how to do it with the following line "Cli_PlcStop"

![image](https://user-images.githubusercontent.com/96547828/197548897-49b8a32d-a7c5-4d8f-b185-e20251087f61.png)

how can i incorporate it in my code to just put my cpu in stop mode because i don't really understand the example you gave me?  thank you

**Reply by swamper123** [reply]
This repo, python-snap7, is just a wrapper around snap7.
The wording/naming of functions, names etc. is not 1:1.

And if you would have clicked on the link to the code I mentioned, you would have recognise this line:

https://github.com/gijzelaerr/python-snap7/blob/076dbee171639b27d7179c9ce69f9444f5669067/snap7/client.py#L100

There is your Cli_PlcStop call (just in Python Context to use the snap7 library).

**Reply by mamax1101** [reply]
Yes I know but I tried the review in the link but I never got it to work...

self._library and self._pointer refers to what?

**Reply by swamper123** [reply]
`self._library` referes to your snap7 library, which should be installed.
`self._pointer` referes to your Client.
But these are things under the hood, you shouldn't worry about, if you just want to use this library.

So what happens if you replace `plc.CpuStop()` with `plc.plc_stop()` in your code?
Any errors? Tracebacks?

**Reply by mamax1101** [reply]
So I put the following code

```
import snap7


plc  = snap7.client.Client()
plc.connect("210.125.55.106", 0, 1)

plc.plc_stop()
```
I executed the code without any problem but my CPU does not change state at all, it stays on.

**Comment by gijzelaerr** [comment]
Closing as answered. The user was calling a non-existent method `CpuStop()` instead of the correct `plc_stop()`.
